### PR TITLE
Add auto documentation generation using sphinx 

### PR DESCRIPTION
**Pull request by ZachEichen**
Addresses issue #34. Adds all the necessary framework and scripts to generate documentation from docstrings using sphinx. 

Documentation functionality was added in commit https://github.com/CODAIT/text-extensions-for-pandas/commit/60a0e5cc663749c99d7a3d987a61c57fb5dff0b3

Travis CI checking of Documentation generation was added in commit https://github.com/CODAIT/text-extensions-for-pandas/commit/4e8661835673c74c721ead6f9739a9d7fd742c4e

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,4 +24,6 @@ install:
 
 
 script:
- - python -m unittest discover
\ No newline at end of file
+ - #check that doc generation is possible
+  - generate_docs.sh
```
Comment (frreiss) [inline .travis.yml:28]: I'm pretty sure you need to say `./generate_docs.sh` or `bash generate_docs.sh` here. Travis build for this PR is currently failing on this line.

File: docs/.gitignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
+* 
+!.gitignore 
+!conf.py
+!index.rst
+!_templates
+!_static 
+!_build
```
Comment (frreiss) [inline docs/.gitignore:7]: Do the `_build`, `_static`, and `_templates` directories need to be checked into git, or will sphinx automatically create them on demand?

File: docs/conf.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+# Configuration file for the Sphinx documentation builder.
```
Comment (frreiss) [inline docs/conf.py:1]: Can you please copy the IBM copyright notice from one of our other `.py` files into the top of this file?

File: docs/index.rst [hunk 4]
```diff
@@ -0,0 +1,15 @@
+Welcome to text_extensions_for_pandas's documentation!
```
Comment (frreiss) [inline docs/index.rst:1]: Should this file be in git, or is it generated on demand?

File: generate_docs.sh [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+# 
+# To make adjustments to the landing page, edit the index.rst file in the docs directory 
+# To make adjustments to the style or change settings, modify the conf.py in the docs directory 
+# 
+# This script is made to run using the same python environment as is used for the package
+# and should be run accordingly 
+
+
+#install sphinx if it is not already installed
+conda install -y sphinx
```
Comment (frreiss) [inline generate_docs.sh:15]: Now that I've merged #38, can you please remove this line from here and add `sphinx` to the dependencies in `config/dev_reqs.txt`?

File: docs/.gitignore (hunk 2 above)
Comment (ZachEichen) [reply docs/.gitignore:7]: they're necessary for sphinx, I'm pretty sure.

File: .travis.yml (hunk 1 above)
Comment (ZachEichen) [reply .travis.yml:28]: oops yeah, I had that fixed on my local version. should be ./

File: docs/index.rst (hunk 4 above)
Comment (ZachEichen) [reply docs/index.rst:1]: this file needs to be in git. its the landing page, and isn't generated automatically.

**Comment by frreiss** [comment]
LGTM. Merging.

**Pull request merged**